### DISCUSSION TITLE
render react-primitives View?

### DISCUSSION THREAD

**Discussion by grahampcharles** [diegomura/react-pdf#1166] [Q&A]
Since the <View> follows the React primitives specification, shouldn't it be possible to render a react-primitives View? Or clone a react-primitives View to a ReactPDF.View?

**Comment by diegomura** [comment]
It's an interesting idea! As you said, this project does follows _some_ of the react primitives specifications, but there are also many that we do not support and ours that react-primitives does not have (specially since 2.0 will be launched with all SVG primitives). So the reusability although possible it will be limited. I should probably get familiarized how react-primitives works and open a PR. Anyone please feel free as well to do it!

**Reply by grahampcharles** [reply]
I tried to write a deep clone, but I don't know the structure well enough; kept running into trouble. It's probably doable, though. The use case was to display a react-primitive View and then render it to a PDF WYSIWYG, like on your REPL. I decided to adopt the same strategy you used -- just using the react-pdf library (the other one) to display the blob that the ReactPDF library (this one) generates.

**Reply by diegomura** [reply]
Sounds good. With the upcoming 2.0 it will be possible to build an HTML PDF viewer so we do not have to either create the PDF or use the other react-pdf to display a preview of the doc.
